src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs

```diff
@@ -45,9 +48,17 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
                     {
                         return new HealthCheckResult(context.Registration.FailureStatus, description: "New QueueClient connection can't be added into dictionary.");
                     }
+
+                    var managementClient = new ManagementClient(_connectionString);
+                    var queueDescription = await managementClient.GetQueueAsync(_queueName);

```

Comment:
What happens if the queue `_queueName` does not exist?

Reply:
MessagingEntityNotFoundException("Queue was not found");

Reply:
Yes, it is also the same exception that will be thrown on a call to queueClient.ScheduleMessageAsync() passing a non-existent queue (Surprisingly, you can create a QueueClient passing in a non-existent queue).
Should I handle a case where there is no queue during the first call, but gets created eventually? Having _requiresSession as static would pose problems in that case.

Reply:
I've updated to handle cases where queue is deleted or created after health check gets run once. This would be useful if this is part of a readiness probe in kubernetes, where the container is not killed on health check failure.

Reply:
> Should I handle a case where there is no queue during the first call, but gets created eventually?

Сontrol will fall into catch block so I think there is no difference.